collector/collector.go

```diff
@@ -0,0 +1,620 @@
+// +build linux
+
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2017 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	utils "github.com/intelsdi-x/snap-plugin-utilities/ns"
+
+	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin"
+
+	"github.com/intelsdi-x/snap-plugin-collector-docker/container"
+	"github.com/intelsdi-x/snap-plugin-collector-docker/container/fs"
+	"github.com/intelsdi-x/snap-plugin-collector-docker/container/network"
+)
+
+const (
+	// namespace vendor prefix
+	PLUGIN_VENDOR = "intel"
+	// namespace plugin name
+	PLUGIN_NAME = "docker"
+	// version of plugin
+	PLUGIN_VERSION = 5
+
+	// each metric starts with prefix "/intel/docker/<docker_id>"
+	lengthOfNsPrefix = 3
+)
+
+var getters map[string]container.StatGetter = map[string]container.StatGetter{
+	"throttling_data": &fs.Cpu{},
+	"cpu_usage":       &fs.CpuAcct{},
+	"cpu_shares":      &fs.CpuShares{},
+	"cache":           &fs.MemoryCache{},
+	"usage":           &fs.MemoryUsage{},
+	"swap_usage":      &fs.SwapMemUsage{},
+	"kernel_usage":    &fs.KernelMemUsage{},
+	"statistics":      &fs.Memory{},
+	"blkio_stats":     &fs.Blkio{},
+	"hugetlb_stats":   &fs.HugeTlb{},
+	"pids_stats":      &fs.Pids{},
+	"cpuset_stats":    &fs.CpuSet{},
+	"network":         &network.Network{},
+	"tcp":             &network.Tcp{},
+	"tcp6":            &network.Tcp6{},
+	"filesystem":      &fs.CpuSet{},
+}
+
+var names map[string]string = map[string]string{
+	"throttling_data": "cpu",
+	"cpu_usage":       "cpuacct",
+	"cpu_shares":      "cpu",
+	"cache":           "memory",
+	"usage":           "memory",
+	"swap_usage":      "memory",
+	"kernel_usage":    "memory",
+	"statistics":      "memory",
+	"blkio_stats":     "blkio",
+	"hugetlb_stats":   "hugetlb",
+	"pids_stats":      "pids",
+	"cpuset_stats":    "cpuset",
+	"spec":            "spec",
+	"network":         "network",
+	"tcp":             "tcp",
+	"tcp6":            "tcp6",
+	"filesystem":      "filesystem",
+}
+
+// New returns initialized docker plugin or error if failed to connect to docker deamon
+func New() (*collector, error) {
+	return &collector{
+		containers: map[string]*container.ContainerData{},
+		mounts:     map[string]string{},
+	}, nil
```

Comment:
After this refactoring, returning error from New() is no longer needed - returned err is always nil. Consider removing error from output and updating go-doc for this function